beginner_source/audio_preprocessing_tutorial.py

```diff
@@ -167,14 +172,125 @@ def normalize(tensor):
 
 print("Median relative difference between original and MuLaw reconstucted signals: {:.2%}".format(err))
 
+######################################################################
+# ``torchaudio`` also supports various ``augmentations``:
+# 
+# -  **TimeStretching**: Wrapping and allowing for batching for the `phase_vocoder` functional.
+# -  **FrequencyMasking**: Apply masks of a desired value to an input spectrogram on frequency.
+# -  **TimeMasking**: Apply masks of a desired value to an input spectrogram on time.
+#
+# An example where this can be used is if a user wants to make a module where the spectrogram is
+# computed and stretched:
+
+num_freqs, hop_length = 400, 512
+model = torch.nn.Sequential(torch.stft(waveform, frame_length=hop_length, fft_size=(num_freqs - 1) * 2),
+                           torchaudio.transforms.TimeStretch(freq=num_freqs, hop_length=hop_length, fixed_rate=1.3),
+                           torchaudio.transforms.ComplexNorm(power=2.0),
+                           torchaudio.transforms.FrequencyMasking(freq_mask_param=60, iid_masks=False),
+                           torchaudio.transforms.TimeMasking(time_mask_param=30, iid_masks=False),
+                           torchaudio.transforms.AmplitudeToDB() )
```

Comment:
This is an example where transforms are combined as neural network layers. It could be interesting to add such an example, but we should highlight it as such. And it could be worth using the term "data augmentation" here, since FrequencyMasking and TimeMasking are adding random variation of the data each time the network is evaluated.

Pragmatically, I'd leave this example for a next iteration unless you are passionate about it :)

Reply:
Let's leave it for the next one... still experimenting with it but it will take more time :)